File: src/main/java/com/my/admin/app/commerce/model/repository/CashRepository.java
package com.my.admin.app.commerce.model.repository;

import com.black.food.manager.model.entity.Cash;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Set;

@Repository
public interface CashRepository extends JpaRepository<Cash, Long> {
    Set<Cash> findByCashStatusId(Long cashStatusId);
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/repository/CashRepository.kt
package com.bk.reactive.app.commerce.admin.my.repository

import com.bk.reactive.app.commerce.admin.my.model.document.Cash
import org.springframework.data.mongodb.repository.Query
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux

@Repository
interface CashRepository : ReactiveMongoRepository<Cash, String> {
    @Query("{'cashStatus._id' : ObjectId(?0)}")
    fun findByCashStatusId(cashStatusId: String): Flux<Cash>

    fun findByStartDateBetween(dateFrom: String?, dateTo: String?): Flux<Cash>
}